### PR TITLE
[grundfosalpha] Initial contribution

### PR DESCRIPTION
**Pull request by tisoft**
This adds support for reading out the data of Grundfos Alpha Pumps with a Grundfos Alpha Reader (https://product-selection.grundfos.com/de/products/alpha-reader/mi401-alpha-reader-98916967?pumpsystemid=2213476387&tab=variant-specifications)

### CODE REVIEW

**Comment by tisoft** [comment]
I had to disable the default filtering of duplicate data in blue, otherwise I could not get the data of the alpha reader. I only used blue on my development laptop, in my production environment I use a bluegiga dongle. That worked out of the box, so the configuration for the blue and bluegiga environment are different by default. The commit in this PR makes the blue implementation behave the same as the bluegiga implementation regarding duplicate data.

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="bluetooth"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<channel-type id="grundfos_flow">
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:7]: Please adapt according to [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention).
```suggestion
	<channel-type id="grundfos-flow">
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		<item-type>Number:VolumetricFlowRate</item-type>
+		<label>Current Flow</label>
+		<description>Current flow</description>
+		<category>flow</category>
+		<state readOnly="true" pattern="%.2f %unit%"/>
+	</channel-type>
+
+	<channel-type id="grundfos_head">
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:15]: Please adapt according to [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention).
```suggestion
	<channel-type id="grundfos-head">
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		<item-type>Number:Length</item-type>
+		<label>Current Head</label>
+		<description>Current Head</description>
+		<category>water</category>
+		<state readOnly="true" pattern="%.2f %unit%"/>
+	</channel-type>
+
+	<channel-type id="grundfos_temperature">
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:23]: Please adapt according to [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention).
```suggestion
	<channel-type id="grundfos-temperature">
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		<item-type>Number:Temperature</item-type>
+		<label>Current Pump Temperature</label>
+		<description>Current Pump Temperature</description>
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:26]:
```suggestion
		<description>Current pump temperature</description>
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+	<thing-type id="mi401">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="roaming"/>
+			<bridge-type-ref id="bluegiga"/>
+			<bridge-type-ref id="bluez"/>
+		</supported-bridge-type-refs>
+
+		<label>Grundfos Alpha Reader MI401</label>
+		<description>An Grundfos Alpha Reader MI401</description>
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:39]:
```suggestion
		<description>A Grundfos Alpha Reader MI401</description>
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+		<category>pump</category>
+
+		<channels>
+			<channel id="rssi" typeId="rssi"/>
+			<channel id="flowRate" typeId="grundfos_flow"/>
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:44]: Please adapt according to [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention).
```suggestion
			<channel id="flow-rate" typeId="grundfos-flow"/>
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+			<channel id="pumpHead" typeId="grundfos_head"/>
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:45]: Please adapt according to [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention).
```suggestion
			<channel id="pump-head" typeId="grundfos-head"/>
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+			<channel id="pumpTemperature" typeId="grundfos_temperature"/>
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:46]: Please adapt according to [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention).
```suggestion
			<channel id="pump-temperature" typeId="grundfos-temperature"/>
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+			<channel id="batteryLevel" typeId="system.battery-level"/>
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:47]: Please adapt according to [naming convention](https://www.openhab.org/docs/developer/guidelines.html#naming-convention).
```suggestion
			<channel id="battery-level" typeId="system.battery-level"/>
```

File: bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+		</channels>
+
+		<config-description>
+			<parameter name="address" type="text">
```
Comment (jlaur) [inline bundles/org.openhab.binding.grundfosalphareader/src/main/resources/OH-INF/thing/thing-types.xml:51]: Perhaps it should be renamed to `macAddress` for clarity?

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks a lot for your contribution! It looks very good. I have added a few initial comments.

Do you know if the protocol for Grundfos Alpha Reader is similar to the one for Grundfos Alpha3, i.e. pumps with built-in Bluetooth support? I have an Alpha3 and would love to help testing and contributing if this could be supported as well. I already did a backport of your binding to 4.0.x to see if might communicate with my pump through BlueZ (Raspberry Pi 3), but unfortunately without success.

In order to be extensible in the future, I would already propose to rename to GrundfosAlpha, i.e. drop "Reader". If Alpha3 support could be added, this could then be a new thing type within the same binding.

I believe the binding id might need to be `org.openhab.binding.bluetooth.grundfosalpha` (i.e. with a bluetooth prefix), but I will need to get back to that to be sure.

Can you also add I18N properties? See https://www.openhab.org/docs/developer/utils/i18n.html#generating-i18n-properties-file.

There are a few compiler warnings and checkstyle warnings to address. You could take a look at target/code-analysis/report.html.

**Comment by tisoft** [comment]
Thanks for the review, I will work on your issues.

Regarding Grundfos Alpha 3:
Support should be possible, since it does use the same BLE implementation. It might use a slightly different protocol. I can't check that myself since I don't have one.
Have you tried the official app from Grundfos to check if that can read out the values?
- https://apps.apple.com/de/app/grundfos-go-balance-pump-tool/id1007147002
- https://play.google.com/store/apps/details?id=com.grundfos.go.balance&hl=de&gl=US

When that works, you could try to dump the bluetooth messages with:

1.	sudo hcitool lescan --duplicate > /dev/null &
2.	sudo hcidump --raw hci

And send me some of the messages, so that I can compare them to mine.

File: bundles/org.openhab.binding.grundfosalphareader/README.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+| flowRate        | Number:VolumetricFlowRate | R          | The flow rate of the pump          |
+| pumpHead        | Number:Length             | R          | The water head above the pump      |
+| pumpTemperature | Number:Temperature        | R          | The temperature of the pump        |
+| batteryLevel    | Number:Percent            | R          | The battery level of the reader    |
+
+## Full Example
+
+_Provide a full usage example based on textual configuration files._
+_*.things, *.items examples are mandatory as textual configuration is well used by many users._
+_*.sitemap examples are optional._
```
[the remark this answers is not in the document]
Comment (tisoft) [reply bundles/org.openhab.binding.grundfosalphareader/README.md:37]: Will add an example as soon as everything else is finalized. :)

**Comment by jlaur** [comment]
> Support should be possible, since it does use the same BLE implementation.

That sounds promising! Just to be clear, support for Alpha3 can be added in another PR as it's clearly out of scope for this one. But I'm very much interested in finally getting my pump online. 😄

> It might use a slightly different protocol. I can't check that myself since I don't have one. Have you tried the official app from Grundfos to check if that can read out the values?

I can read a lot of data with Grundfos Go Remote:
https://play.google.com/store/apps/details?id=com.grundfos.go

> * https://play.google.com/store/apps/details?id=com.grundfos.go.balance&hl=de&gl=US

I tried to install this one, but already at first step it seemed like it would start some major calibration, so didn't dare to go further.

> When that works, you could try to dump the bluetooth messages with:
> 
> 1. sudo hcitool lescan --duplicate > /dev/null &
> 2. sudo hcidump --raw hci
> 
> And send me some of the messages, so that I can compare them to mine.

Thanks. I've sent some messages by private mail.

Perhaps this might also be helpful, I had a brief look at that when previously collecting messages and trying to figure them out:
https://github.com/esphome/esphome/tree/dev/esphome/components/alpha3

**Comment by tisoft** [comment]
I have renamed/moved the add-on to org.openhab.binding.bluetooth.grundfosalpha.

Could you help me with the following warnings? What do I have to do to fix them?

[WARNING] OH-INF.thing.thing-types.xml:[0]
Missing the supported bridge with id roaming
[WARNING] OH-INF.thing.thing-types.xml:[0]
Missing the supported bridge with id bluegiga
[WARNING] OH-INF.thing.thing-types.xml:[0]
Missing the supported bridge with id bluez

I had a quick look on your Alpha3 logs and I think they can be supported.

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/README.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+## Channels
+
+| Channel         | Type                      | Read/Write | Description                        |
+|-----------------|---------------------------|------------|------------------------------------|
+| rssi            | Number                    | R          | Received Signal Strength Indicator |
+| flowRate        | Number:VolumetricFlowRate | R          | The flow rate of the pump          |
+| pumpHead        | Number:Length             | R          | The water head above the pump      |
+| pumpTemperature | Number:Temperature        | R          | The temperature of the pump        |
+| batteryLevel    | Number:Percent            | R          | The battery level of the reader    |
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/README.md:31]: Renamed to:
```suggestion
| flow-rate         | Number:VolumetricFlowRate | R          | The flow rate of the pump          |
| pump-head        | Number:Length             | R          | The water head above the pump      |
| pump-temperature | Number:Temperature        | R          | The temperature of the pump        |
| battery-level    | Number:Percent            | R          | The battery level of the reader    |
```

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderHandlerFactory.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * The {@link GrundfosAlphaReaderHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Markus Heberling - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.grundfosalphareader", service = ThingHandlerFactory.class)
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderHandlerFactory.java:35]:
```suggestion
@Component(configurationPid = "binding.bluetooth.grundfosalpha", service = ThingHandlerFactory.class)
```

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/addon/addon.xml
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/addon/addon.xml:1]: I'm not sure if this is needed anymore when adopted as a Bluetooth extension. It looks like the other bindings do not have it.

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/addon/addon.xml [hunk 14]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<addon:addon id="grundfosalpha" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:addon="https://openhab.org/schemas/addon/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/addon/v1.0.0 https://openhab.org/schemas/addon-1.0.0.xsd">
+
+	<type>binding</type>
+	<name>GrundfosAlpha Binding</name>
+	<description>This is the binding for GrundfosAlpha.</description>
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/addon/addon.xml:8]:
```suggestion
	<description>This is the binding for Grundfos Alpha circulator pumps.</description>
```

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+		<label>Current Flow</label>
+		<description>Current flow</description>
+		<category>flow</category>
+		<state readOnly="true" pattern="%.2f %unit%"/>
+	</channel-type>
+
+	<channel-type id="grundfos-head">
+		<item-type>Number:Length</item-type>
+		<label>Current Head</label>
+		<description>Current Head</description>
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml:18]:
```suggestion
		<description>Current head</description>
```

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+<thing:thing-descriptions bindingId="bluetooth"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<channel-type id="grundfos-flow">
+		<item-type>Number:VolumetricFlowRate</item-type>
+		<label>Current Flow</label>
+		<description>Current flow</description>
+		<category>flow</category>
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml:11]: We usually provide them with first letter in uppercase:
```suggestion
		<category>Flow</category>
```

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+		<state readOnly="true" pattern="%.2f %unit%"/>
+	</channel-type>
+
+	<channel-type id="grundfos-head">
+		<item-type>Number:Length</item-type>
+		<label>Current Head</label>
+		<description>Current Head</description>
+		<category>water</category>
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml:19]:
```suggestion
		<category>Water</category>
```

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+		<state readOnly="true" pattern="%.2f %unit%"/>
+	</channel-type>
+
+	<channel-type id="grundfos-temperature">
+		<item-type>Number:Temperature</item-type>
+		<label>Current Pump Temperature</label>
+		<description>Current pump temperature</description>
+		<category>temperature</category>
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml:27]: You could also provide semantic tags here:
```suggestion
		<category>Temperature</category>
		<tags>
			<tag>Measurement</tag>
			<tag>Temperature</tag>
		</tags>
```

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+	<thing-type id="mi401">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="roaming"/>
+			<bridge-type-ref id="bluegiga"/>
+			<bridge-type-ref id="bluez"/>
+		</supported-bridge-type-refs>
+
+		<label>Grundfos Alpha Reader MI401</label>
+		<description>A Grundfos Alpha Reader MI401</description>
+		<category>pump</category>
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/thing/thing-types.xml:40]:
```suggestion
		<category>Pump</category>
```

File: bundles/pom.xml [hunk 20]
```diff
@@ -77,6 +77,7 @@
     <module>org.openhab.binding.bluetooth.daikinmadoka</module>
     <module>org.openhab.binding.bluetooth.enoceanble</module>
     <module>org.openhab.binding.bluetooth.generic</module>
+    <module>org.openhab.binding.bluetooth.grundfosalpha</module>
```
Comment (jlaur) [inline bundles/pom.xml:80]: One down for alphabetizing. 🙂

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderHandlerFactory.java
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderHandlerFactory.java:1]: This file should probably be renamed to `GrundfosAlphaHandlerFactory`.

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderBindingConstants.java
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderBindingConstants.java:1]: This file should probably be renamed to `GrundfosAlphaBindingConstants`.

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderDiscoveryParticipant.java
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderDiscoveryParticipant.java:1]: This file should probably be renamed to `GrundfosAlphaDiscoveryParticipant`.

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderDiscoveryParticipant.java [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This discovery participant is able to recognize enoceanble devices and create discovery results for them.
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaReaderDiscoveryParticipant.java:35]:
```suggestion
 * This discovery participant is able to recognize Grundfos Alpha devices and create discovery results for them.
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Not much found on second pass, only some minor comments.

**Comment by jlaur** [comment]
> I have renamed/moved the add-on to org.openhab.binding.bluetooth.grundfosalpha.

Great!

> Could you help me with the following warnings? What do I have to do to fix them?
> 
> [WARNING] OH-INF.thing.thing-types.xml:[0] Missing the supported bridge with id roaming [WARNING] OH-INF.thing.thing-types.xml:[0] Missing the supported bridge with id bluegiga [WARNING] OH-INF.thing.thing-types.xml:[0] Missing the supported bridge with id bluez

Sure, I will have a look.

> I had a quick look on your Alpha3 logs and I think they can be supported.

That would be awesome, let me know how I can help. 🙂

**Comment by jlaur** [comment]
> > [WARNING] OH-INF.thing.thing-types.xml:[0] Missing the supported bridge with id roaming [WARNING] OH-INF.thing.thing-types.xml:[0] Missing the supported bridge with id bluegiga [WARNING] OH-INF.thing.thing-types.xml:[0] Missing the supported bridge with id bluez
> 
> Sure, I will have a look.

I think you are missing these in the thing type definition:
```xml
		<supported-bridge-type-refs>
			<bridge-type-ref id="roaming"/>
			<bridge-type-ref id="bluegiga"/>
			<bridge-type-ref id="bluez"/>
		</supported-bridge-type-refs>
```

**Comment by jlaur** [comment]
@tisoft - just in case you missed them, there are still a few small comments to resolve. We can merge right after that. :)

**Comment by tisoft** [comment]
Yeah I know. Was a bit busy with work the last days. Will get back to this as soon as possible.

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaDiscoveryParticipant.java [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+ *
+ */
+@NonNullByDefault
+@Component
+public class GrundfosAlphaDiscoveryParticipant implements BluetoothDiscoveryParticipant {
+    private final Logger logger = LoggerFactory.getLogger(GrundfosAlphaDiscoveryParticipant.class);
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(GrundfosAlphaBindingConstants.THING_TYPE_MI401);
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/java/org/openhab/binding/bluetooth/grundfosalpha/internal/GrundfosAlphaDiscoveryParticipant.java:47]:
```suggestion
        return Set.of(GrundfosAlphaBindingConstants.THING_TYPE_MI401);
```

File: bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/i18n/grundfosalpha.properties [hunk 23]
```diff
@@ -0,0 +1,23 @@
+# add-on
+
+addon.grundfosalpha.name = GrundfosAlpha Binding
+addon.grundfosalpha.description = This is the binding for Grundfos Alpha circulator pumps.
+
```
Comment (jlaur) [inline bundles/org.openhab.binding.bluetooth.grundfosalpha/src/main/resources/OH-INF/i18n/grundfosalpha.properties:5]: This should probably be removed after the removal of the addon.xml file. If you then regenerate this file (see
https://www.openhab.org/docs/developer/utils/i18n.html#generating-i18n-properties-file), the changed text for `channel-type.grundfosalpha.grundfos-head.description` below should also be updated.

File: bundles/org.openhab.binding.grundfosalphareader/README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (jlaur) [reply bundles/org.openhab.binding.grundfosalphareader/README.md]: Besides two minor comments I've just added, everything else now seems finalized. 🙂

**Comment by jlaur** [comment]
@kaikreuzer - can you add @tisoft as contributor so we will be automatically assigned as code reviewer for the new Grundfos Alpha binding? 🙂

**Comment by kaikreuzer** [comment]
Sure, done!

**Pull request merged**